lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -267,54 +254,70 @@
-                                            teamMemberships: self.teamMemberships,
-                                            maxTeamSize: self.maxTeamSize,
-                                            teamParams: {
-                                                courseID: self.courseID,
-                                                topicID: topic.get('id'),
-                                                teamsUrl: self.teamsUrl,
-                                                topicName: topic.get('name'),
-                                                languages: self.languages,
-                                                countries: self.countries,
-                                                teamsDetailUrl: self.teamsDetailUrl
-                                            }
+                                            showSortControls: true
                                         });
-                                        deferred.resolve(
-                                            self.createViewWithHeader(
-                                                {
-                                                    mainView: teamsView,
-                                                    subject: topic
-                                                }
-                                            )
-                                        );
+                                        deferred.resolve(teamsView);
                                     });
                             });
                     }
                     return deferred.promise();
                 },
 
+                createTeamsListView: function(options) {
+                    var topic = options.topic,
+                        collection = options.collection,
+                        teamsView = new TopicTeamsView({
+                            router: this.router,
+                            context: this.context,
+                            model: topic,
+                            collection: collection,
+                            teamMemberships: this.teamMemberships,
+                            showSortControls: options.showSortControls
+                        }),
+                        searchFieldView = new SearchFieldView({
+                            type: 'teams',
+                            label: gettext('Search teams'),
+                            collection: collection
+                        }),
+                        viewWithHeader = this.createViewWithHeader({
+                            subject: topic,
+                            mainView: teamsView,
+                            headerActionsView: searchFieldView,
+                            title: options.title,
+                            description: options.description,
+                            breadcrumbs: options.breadcrumbs
+                        });
+                    this.listenTo(collection, 'sync', function() {
```

Comment:
But when will the collection by synced and yet we still want to maintain the search? Can you explain the reproduction steps that would result in a sync?